### DISCUSSION TITLE
mise: support backends

### DISCUSSION THREAD

**Discussion by risu729** [renovatebot/renovate#30282] [Suggest an Idea]
### Tell us more.

mise has a backend feature to use npm, Cargo, Pipx, and more to install CLIs.
https://mise.jdx.dev/dev-tools/backends/

Though it's still experimental, it would be great if Renovate supports it. I'm currently using regex managers to update them.

minimal reproduction repo: https://github.com/risu729/renovate-mise-backends

e.g.
`.mise.toml`

```toml
[tools]
"go:github.com/DarthSim/hivemind" = "1.0.6"
"cargo:eza" = "0.18.21"
"npm:prettier" = "3.3.2"
"pipx:psf/black" = "24.4.1"
"pipx:yamllint" = "1.35.0"
"spm:tuist/tuist" = "4.21.1"
"ubi:goreleaser/goreleaser" = "v2.0.1"
```

**Comment by secustor** [comment]
Any specific backend you are after? 

These would be pretty easy:
- go
- npm
- cargo
I can see these added as quick win until that feature is no longer experimental.

And then potentially additional issues for other backends, as these require more work.

**Comment by risu729** [comment]
Thanks for your reply.

I'm currently using npm, cargo, and ubi backend.
e.g. https://github.com/risu729/dotfiles/blob/2e62c94505958df2aa599553c6b2de11ed35f115/mise.toml#L4-L32

For [ubi](https://github.com/houseabsolute/ubi), it might be simple to implement because the datasource is GitHub releases or tags.

However, it still requires some work to support some monorepo packages, because we need to filter tags matching prefixes.
```toml
# renovate: extractVersion=^cli/v(?<version>.+)
"ubi:biomejs/biome" = "cli/v1.8.3"
```

Just a note, I'm updating them using regex managers.
https://github.com/risu729/dotfiles/blob/2e62c94505958df2aa599553c6b2de11ed35f115/renovate.json5#L18-L81

**Reply by secustor** [reply]
If only an extractVersion has to set that would be fine for me too.

**Reply by rarkins** [reply]
Should this be converted to a single feature request issue, or multiple?

**Reply by secustor** [reply]
multiple

One for: 
- go
- npm
- cargo

and another for ubi as it is more complex 

the rest, when the feature leaves experimental

**Reply by risu729** [reply]
I think there are no other configurations we need to consider.
mise also supports URL syntax for ubi, like `ubi:https://github.com/goreleaser/goreleaser/releases/download/v1.16.2/goreleaser_Darwin_arm64.tar.gz`, but Renovate can ignore it.
https://mise.jdx.dev/dev-tools/backends/ubi.html#supported-ubi-syntax

**Comment by secustor** [comment]
Created a tracking issue https://github.com/renovatebot/renovate/issues/30387
